### PR TITLE
Use npm instead of yarn and create a Dockerfile.test

### PR DESCRIPTION
**Pull request by zhuje**

### CODE REVIEW

File: Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM registry.redhat.io/ubi9/nodejs-18:latest AS web-builder
+
+WORKDIR /usr/src/app
+
+RUN npm install --global yarn
```
[the remark this answers is not in the document]
Comment (zhuje) [reply Dockerfile.test]: Updated in the latest commit.

**Review by jgbernalp — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,9 @@
 FROM registry.access.redhat.com/ubi8/nodejs-16:latest AS build
 USER root
-RUN command -v yarn || npm i -g yarn
 
 ADD . /usr/src/app
```
[the remark this answers is not in the document]
Comment (zhuje) [reply Dockerfile:4]: In the next commit I'll update this to : 
```
COPY package*.json .
RUN npm install

ADD . /usr/src/app
RUN npm run build
```

File: Dockerfile [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-RUN command -v yarn || npm i -g yarn
 
-ADD . /usr/src/app
 WORKDIR /usr/src/app
-RUN yarn install && yarn build
+
+COPY package*.json .
+RUN npm install
+
+ADD . /usr/src/app
```
Comment (jgbernalp) [inline Dockerfile:9]: If the workdir is already defined we don't need to copy the whole path. Also the frontend app should be contained within the `web` folder
```suggestion
ADD web/ web/
```

**Review by jgbernalp — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (zhuje) [reply Dockerfile]: Moved frontend app into /web folder in the latest commit.

File: Dockerfile.test [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FROM registry.redhat.io/ubi9/nodejs-18:latest AS web-builder
+
+WORKDIR /usr/src/app
+
+USER 0
+
+COPY package*.json .
```
Comment (jgbernalp) [inline Dockerfile.test:7]: This also needs to consider the `web` path

File: web/package.json [hunk 5]
```diff
@@ -6,10 +6,10 @@
...
   "scripts": {
     "clean": "rm -rf dist",
-    "build": "yarn clean && NODE_ENV=production yarn ts-node node_modules/.bin/webpack",
-    "build-dev": "yarn clean && yarn ts-node node_modules/.bin/webpack",
-    "start": "yarn ts-node node_modules/.bin/webpack serve",
-    "start-console": "./start-console.sh",
+    "build": "npm run && NODE_ENV=production npm run ts-node node_modules/.bin/webpack",
+    "build-dev": "npm run clean && npm run ts-node node_modules/.bin/webpack",
+    "start": "npm run ts-node node_modules/.bin/webpack serve",
+    "start-console": "cd .. && ./scripts/start-console.sh",
```
Comment (jgbernalp) [inline web/package.json:12]:
```suggestion
    "start-console": "../scripts/start-console.sh",
```

**Review by jgbernalp — CHANGES_REQUESTED** [review]

File: Dockerfile.test (hunk 4 above)
Comment (zhuje) [reply Dockerfile.test]: Good point. I updated this in the latest commit.

**Review by jgbernalp — CHANGES_REQUESTED** [review]

**Review by jgbernalp — APPROVED** [review]

**Pull request merged**